packages/component/src/Utils/activityAltText.ts

```diff
@@ -0,0 +1,40 @@
+import isSSML from './isSSML';
+
+/** Retrieves all text nodes from a given DOM tree as flattened array. */
+function allTextContents(element: Node): string[] {
+  const nodes: Node[] = [].slice.call(element.childNodes);
+  const results: string[] = [];
+
+  while (nodes.length) {
+    const { childNodes, nodeType, textContent } = nodes.shift();
+
+    if (nodeType === Node.TEXT_NODE) {
+      const trimmedTextContent = textContent.trim();
+
+      trimmedTextContent && results.push(trimmedTextContent);
+    } else {
+      nodes.unshift(...[].slice.call(childNodes));
+    }
+  }
+
+  return results;
+}
+
+/** Returns the alt text for an activity. */
+export default function activityAltText(activity: any): string {
```

Comment:
I've done typings for the bf schema in my playground work - I would recommend porting that from the playground & expanding as needed, as this project moves towards TS.

Reply:
I think our official typings are at https://github.com/microsoft/botbuilder-js/tree/main/libraries/botframework-schema. Unsure if it works in a browser.